Talend/ui
# chore(ARCH-404/CI): decrease time needed

**What is the problem this PR is trying to solve?**

CI takes 20 minutes to execute. Let's try to improve the situation.

learna ci (tests):
* install (2 to 3 minutes)
* test (7 minutes)
* size check (10 minutes)

pr-demo (surge)
* install (2 to 3 minutes)
* pre-release (3 minutes)
* test:demo (15 minutes)

test:demo times
```
 #### RUNNER: yarn workspaces --silent info exit code 0 in 0.301 seconds
#### RUNNER: yarn workspace --silent @talend/icons run test:demo exit code 0 in 0.706 seconds
#### RUNNER: yarn workspace --silent @talend/design-tokens run test:demo exit code 0 in 35.821 seconds
#### RUNNER: yarn workspace --silent @talend/design-system run test:demo exit code 0 in 121.406 seconds
#### RUNNER: yarn workspace --silent @talend/react-components run test:demo exit code 0 in 89.989 seconds
#### RUNNER: yarn workspace --silent @talend/react-stepper run test:demo exit code 0 in 79.884 seconds
#### RUNNER: yarn workspace --silent @talend/react-forms run test:demo exit code 0 in 96.093 seconds
#### RUNNER: yarn workspace --silent @talend/react-faceted-search run test:demo exit code 0 in 119.494 seconds
#### RUNNER: yarn workspace --silent @talend/react-dataviz run test:demo exit code 0 in 112.64 seconds
#### RUNNER: yarn workspace --silent @talend/react-cmf run test:demo exit code 0 in 23.402 seconds
#### RUNNER: yarn workspace --silent @talend/react-datagrid run test:demo exit code 0 in 86.92 seconds
#### RUNNER: yarn workspace --silent @talend/react-containers run test:demo exit code 0 in 106.981 seconds
```



**What is the chosen solution to this problem?**

* add option to choose if executed in // in workspace script
* rewrite size check to commit a file (keeping threshold to 100 bytes) (Minus 10 minutes on test ci)
* move playground outside of test:demo so pre-release and test:demo can be run in // (minus 5 minutes on pr-demo)
* enforce cache on sub node_modules which contains the .cache for webpack (should decrease by two) so test:demo took 5 minutes !


learna ci (tests):
* install (2 to 3 minutes)
* test (7 minutes)
* size check (0 minute)

pr-demo (surge)
* install (2 to 3 minutes)
* pre-release (3 minutes)
* test:demo (5 minutes)

```
 #### RUNNER: yarn workspaces --silent info exit code 0 in 0.254 seconds
#### RUNNER: yarn workspace --silent @talend/icons run test:demo exit code 0 in 0.581 seconds
#### RUNNER: yarn workspace --silent @talend/design-tokens run test:demo exit code 0 in 13.541 seconds
#### RUNNER: yarn workspace --silent @talend/design-system run test:demo exit code 0 in 58.639 seconds
#### RUNNER: yarn workspace --silent @talend/react-components run test:demo exit code 0 in 29.064 seconds
#### RUNNER: yarn workspace --silent @talend/react-stepper run test:demo exit code 0 in 26.548 seconds
#### RUNNER: yarn workspace --silent @talend/react-forms run test:demo exit code 0 in 29.213 seconds
#### RUNNER: yarn workspace --silent @talend/react-faceted-search run test:demo exit code 0 in 45.533 seconds
#### RUNNER: yarn workspace --silent @talend/react-dataviz run test:demo exit code 0 in 37.081 seconds
#### RUNNER: yarn workspace --silent @talend/react-cmf run test:demo exit code 0 in 19.359 seconds
#### RUNNER: yarn workspace --silent @talend/react-datagrid run test:demo exit code 0 in 28.261 seconds
#### RUNNER: yarn workspace --silent @talend/react-containers run test:demo exit code 0 in 32.042 seconds
```

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR



**[ ] This PR introduces a breaking change**
